pyccl/baryons/baccoemu_baryons.py

```diff
@@ -0,0 +1,136 @@
+        
+        self.bcm_params = {
+            'M_c'           : log10_M_c,
+            'eta'           : log10_eta,
+            'beta'          : log10_beta,
+            'M1_z0_cen'     : log10_M1_z0_cen,
+            'theta_out'     : log10_theta_out,
+            'theta_inn'     : log10_theta_inn,
+            'M_inn'         : log10_M_inn
+        }
+    
+    def _sigma8tot_2_sigma8cold(self, emupars, sigma8tot):
+        if hasattr(emupars['omega_cold'], '__len__'):
+            _emupars = {}
+            for pname in emupars:
+                _emupars[pname] = emupars[pname][0]
+        else:
+            _emupars = emupars
+        A_s_fid = 2.1e-9
+        sigma8tot_fid = self.mpk.get_sigma8(cold=False, A_s=A_s_fid, **_emupars)
+        A_s = (sigma8tot / sigma8tot_fid)**2 * A_s_fid
+        return self.mpk.get_sigma8(cold=True, A_s=A_s, **_emupars)
+
+    def boost_factor(self, cosmo, k, a):
+        """The baccoemu BCM model boost factor for baryons.
+
+        Args:
+            cosmo (:class:`~pyccl.cosmology.Cosmology`): Cosmological parameters.
+            k (:obj:`float` or `array`): Wavenumber (in :math:`{\\rm Mpc}^{-1}`).
+            a (:obj:`float` or `array`): Scale factor.
+
+        Returns:
+            :obj:`float` or `array`: Correction factor to apply to
+                the power spectrum.
+        """ # noqa
+
+        if hasattr(a, '__len__'):
+            emupars = dict(
+                omega_cold = np.full((len(a)), cosmo['Omega_c'] + cosmo['Omega_b']),
+                omega_baryon = np.full((len(a)), cosmo['Omega_b']),
+                ns = np.full((len(a)), cosmo['n_s']),
+                hubble = np.full((len(a)), cosmo['h']),
+                neutrino_mass = np.full((len(a)), np.sum(cosmo['m_nu'])),
+                w0 = np.full((len(a)), cosmo['w0']),
+                wa = np.full((len(a)), cosmo['wa']),
+                expfactor = a
+            )
+        else:
+            emupars = dict(
+                omega_cold = cosmo['Omega_c'] + cosmo['Omega_b'],
+                omega_baryon = cosmo['Omega_b'],
+                ns = cosmo['n_s'],
+                hubble = cosmo['h'],
+                neutrino_mass = np.sum(cosmo['m_nu']),
+                w0 = cosmo['w0'],
+                wa = cosmo['wa'],
+                expfactor = a
+            )
+
+        if np.isnan(cosmo['A_s']):
+            sigma8tot = cosmo['sigma8']
+            sigma8cold = self._sigma8tot_2_sigma8cold(emupars, sigma8tot)
+            if hasattr(a, '__len__'):
+                emupars['sigma8_cold'] = np.full((len(a)), sigma8cold)
+            else:
+                emupars['sigma8_cold'] = sigma8cold
+        else:
+            if hasattr(a, '__len__'):
+                emupars['A_s'] = np.full((len(a)), cosmo['A_s'])
+            else:
+                emupars['A_s'] = cosmo['A_s']
+
+        _, fka = self.mpk.get_baryonic_boost(k=k / cosmo['h'], **{**emupars, **self.bcm_params})
```

Comment:
just so I understand, bacco has its own internal interpolator that will get you the boost factor at whatever k you ask, right? What's the first output variable we're ignoring here, out of curiosity?

Reply:
Yep, it always returns a tuple with (k, quantity), in this case quantity being the baryonic boost. If you don't specify k as a keyword argument, you get the original k-array on which the emu is defined, otherwise you get the interpolated version. Since here I pass k I ignore the first output variable.